sources/packages/backend/libs/integrations/src/services/ssh/sftp-integration-base.ts

```diff
@@ -154,10 +155,29 @@ export abstract class SFTPIntegrationBase<DownloadType> {
           return false;
         }
       }
-      // Read all the file content and create a buffer with 'ascii' encoding.
-      const fileContent = await client.get(remoteFilePath, undefined, {
-        readStreamOptions: { encoding: FILE_DEFAULT_ENCODING },
-      });
+      let fileContent: string | NodeJS.WritableStream | Buffer;
+      const fileExtension = path.parse(remoteFilePath).ext.toLocaleLowerCase();
+      const isZIPFile = fileExtension === ".zip";
+      if (isZIPFile) {
+        // Read the zip file content with null encoding to avoid data corruption.
+        const compressedFileContent = (await client.get(
+          remoteFilePath,
+          undefined,
+          { readStreamOptions: { encoding: null } },
+        )) as Buffer;
+        // Read the first file content with 'ascii' encoding.
+        const { fileName, data } = readFirstExtractedFile(
+          compressedFileContent,
+          { encoding: FILE_DEFAULT_ENCODING },
+        );
+        this.logger.log(`Extracted the first file ${fileName}.`);
+        fileContent = data;
+      } else {
+        // Read all the file content and create a buffer with 'ascii' encoding.
+        fileContent = await client.get(remoteFilePath, undefined, {
+          readStreamOptions: { encoding: FILE_DEFAULT_ENCODING },
+        });
```

Comment:
Similar to the compressed get at the line 167 the below can be cast.
```ts
 // Read all the file content and create a buffer with 'ascii' encoding.
fileContent = (await client.get(remoteFilePath, undefined, {
  readStreamOptions: { encoding: FILE_DEFAULT_ENCODING },
})) as string;
```
This would allow the fileContent to be declared as string (which really is) instead of `let fileContent: string | NodeJS.WritableStream | Buffer;` which can be misleading.